pkg/controller/logstorage/initializer/conditions_controller.go

```diff
@@ -90,18 +91,272 @@ func (r *LogStorageConditions) Reconcile(ctx context.Context, request reconcile.
 		return reconcile.Result{}, err
 	}
 
-	// Update conditions on the LogStorage object based on the status of the various components.
-	// TODO: Incorporate status from other sub controllers as well.
-	ts := &operatorv1.TigeraStatus{}
-	if err := r.client.Get(ctx, types.NamespacedName{Name: TigeraStatusName}, ts); err != nil {
+	// Fetch the current status condition for log storage
+	currentCondition := getCurrentConditions(ls.Status.Conditions)
+
+	// Aggregate TigeraStatus conditions from all logstorage subcontrollers into a map,
+	// using the condition type (e.g., Available, Progressing, and Degraded) as the key.
+	desiredConditions, err := r.getDesiredConditions(ctx)
+	if err != nil {
 		return reconcile.Result{}, err
 	}
 
-	ls.Status.Conditions = status.UpdateStatusCondition(ls.Status.Conditions, ts.Status.Conditions)
+	// Compare and update the current StatusCondition if there are any new changes
+	ls.Status.Conditions = updateConditions(currentCondition, desiredConditions)
+
 	if err := r.client.Status().Update(ctx, ls); err != nil {
 		log.WithValues("reason", err).Info("Failed to update LogStorage status conditions")
 		return reconcile.Result{}, err
 	}
 
 	return reconcile.Result{}, nil
 }
+
+func getCurrentConditions(lsConditions []metav1.Condition) map[string]metav1.Condition {
+	statusCondition := make(map[string]metav1.Condition)
+	for _, condition := range lsConditions {
+		statusCondition[condition.Type] = condition
+	}
+	return statusCondition
+}
+
+// getDesiredConditions checks sub-controller TigeraStatus objects and builds the desired
+// conditions based on the health of each. It returns a map for easy lookup - the key is the condition type.
+func (r *LogStorageConditions) getDesiredConditions(ctx context.Context) (map[string]metav1.Condition, error) {
+
+	desiredTigeraStatusMap := make(map[operatorv1.StatusConditionType]operatorv1.TigeraStatusCondition)
+
+	logStorageInstances := []string{TigeraStatusName, TigeraStatusLogStorageAccess, TigeraStatusLogStorageElastic, TigeraStatusLogStorageSecrets}
+	if r.multiTenant {
+		logStorageInstances = append(logStorageInstances, TigeraStatusLogStorageUsers)
+	} else {
+		logStorageInstances = append(logStorageInstances, TigeraStatusLogStorageESMetrics, TigeraStatusLogStorageKubeController)
+	}
+
+	for _, logStorage := range logStorageInstances {
```

Comment:
@vara2504 I think this particular function can probably still be simplified. The thing I'd like to avoid is the many layers of helper function that modify the conditions. I think we should be able to factor this so that we build each condition exactly once in once place. Let me know what you think of logic like this:

```go

// Keep track of which instances are in which state.
states := map[string][]string {
  ComponentDegraded: []string{},
  ComponentAvailable: []string{},
  ComponentProgressing: []string{},
}

// We also keep track of the oldest observed generation here so we can add it to the conditions later.
var observedGeneration int64

// Build up the lists of which components are in which state.
for _, instance := range expectedInstances {
  ts := &operatorv1.TigeraStatus{}
  if err := r.client.Get(ctx, namespacedName, ts); errors.IsNotFound(err) {
    // Expected status not found, treat it as an error.
    states[ComponentDegraded] = append(states[ComponentDegraded], instance.Name)
    continue
  } else if err != nil {
    return nil, err
  }

  // Found it - add to states map, which we use to track which TigeraStatuses are in which state(s).
  for _, condition := range ts.Status.Conditions {
    if condition.Status == ConditionTrue {
      states[condition.Status] = append(states[condition.Status], instance.Name)
    }
    if observedGeneration == 0 || condition.ObservedGeneration < observedGeneration {
      observedGeneration = condition.ObservedGeneration
    }
  }
}

// Convert the states map to a set of conditions.
conditions := map[string]metav1.Condition{}
for _, statusType, instances := range states {

  condition := metav1.Condition{
    Type: statusType, 
    ObservedGeneration: observedGeneration,
  }  

  if statusType != ComponentAvailable && len(instances) != 0 {
    // This condition is true.
    condition.Status = operatorv1.ConditionTrue
    conditoin.Reason = operatorv1.NotReady
    condition.Message = fmt.Sprintf("The following sub-controllers are in this condition: %+v", instances)
  } else if statusType != ConditionAvailable {
    // This condition is false.
    condition.Status = operatorv1.ConditionFalse
  } else {
    // This is the available condition, which is only true if all statuses are available.
    if len(instances) == len(expectedInstances) {
      condition.Status = operatorv1.ConditionTrue
      condition.Message = "All sub-controllers are available"
    } else {
      condition.Status = operatorv1.ConditionFalse
    }
  }

  // Store the condition.
  conditions[statusType] = condition
} 
return conditions
```

Reply:
I think this should remove the need for a number of the helper functions, like:

- transformIntoLogStorageConditions
- setAvailable, setProgressing, setDegraded
- clearAvailable, clearProgressing, clearDegraded
- mergeCondition

As always, I could be missing an edge case though so let me know if this is too simple for what we need!